Frameworks/UIKit/UISearchBar.mm

```diff
@@ -72,18 +73,39 @@ - (BOOL)resignFirstResponder {
  @Status Interoperable
 */
 - (void)setPrompt:(NSString*)prompt {
-    if (_promptLabel == nil) {
-        CGRect promptFrame = CGRectMake(0, c_marginTopForPrompt, self.frame.size.width, c_defaultTextFieldHeight);
-
-        self->_promptLabel.attach([[UILabel alloc] initWithFrame:promptFrame]);
-        [self->_promptLabel setTextAlignment:UITextAlignmentCenter];
-        [self->_promptLabel setBackgroundColor:nil];
-        [self->_promptLabel setTextColor:[UIColor blackColor]];
-        [self addSubview:self->_promptLabel];
+    if ((([_prompt length] == 0) && ([prompt length] == 0)) || [_prompt isEqualToString:prompt]) {
+        // if old prompt text is "equal" to the new prompt text (nil and emptyString are considered equal), quick return
+        return;
+    }
+
+    // otherwise, old prompt text is different from new prompt text, need either add or remove prompt label
+    _prompt = [prompt copy];
```

Comment:
IIRC copy will return an object with a retain count of 1, so this similar to assigning a StrongId<NSString> = [[NSString alloc] init] (which ends up bumping the retain count to 2), thus still leaking the pointer.  I'd confirm with Dustin or step through the debugger, but I *think* you want to .attach here to avoid the extra refcount bump.